### PR TITLE
Remove CarbonIterator

### PR DESCRIPTION
**Pull request by jackylk**
CarbonIterator is exactly same as java.util.Iterator, use java.util.Iterator instead

### CODE REVIEW

**Comment by jackylk** [comment]
retest this please

**Comment by kumarvishal09** [comment]
Carbon Iterator was added so no need to implement remove method, we can have abstract class which will implement iterator interface and all the concrete classes will extends the abstract class. Abstract class can have remove method implementation, so we can avoid duplicate remove method from all the concrete implementation

**Comment by jackylk** [comment]
@kumarvishal09 modified

**Comment by ravipesala** [comment]
@jackylk  Please rebase the PR.

**Comment by jackylk** [comment]
@ravipesala rebased

File: core/src/main/java/org/carbondata/core/iterator/PlainIterator.java [hunk 1]
```diff
@@ -19,20 +19,20 @@
...
-   * @return <tt>true</tt> if the iterator has more elements.
-   */
-  boolean hasNext();
+import java.util.Iterator;
+
+/**
+ * PlainIterator adds default implement for remove. This is required for Java 7.
+ * @param <E>
+ */
+public abstract class PlainIterator<E> implements Iterator<E> {
```
Comment (ravipesala) [inline core/src/main/java/org/carbondata/core/iterator/PlainIterator.java]: Can you name as `CarbonIterator`, so that changes will be very minimal.

**Comment by ravipesala** [comment]
LGTM. Merged to master.

**Pull request merged**